core/IO.carp

```diff
@@ -72,9 +74,47 @@
                                (IO.fclose f)
                                (Result.Success (String.from-chars &r)))))))
 
+  (doc      unsafe-read-file "returns the contents of a file passed as argument as a string. Note: there is no way to distinguish the output for an empty file and a missing file!")
+  (register unsafe-read-file (Fn [&String] String))
+  (doc  read-file "Reads the content of a file into a (Result String String).\nIt is intended for text files, since the way to determine the length of a String is to use strlen() which probably will be inaccurate for binaries.")
+  (defn read-file [filename]
+    (let [  finp?  (open-file filename "rb") ]
+      (if (Result.error? &finp?)
+        (Result.Error (fmt "Failed to open file='%s', error-number=%d" filename (Result.unsafe-from-error finp?) ))
+        (let [  finput  (Result.unsafe-from-success finp?)
+                length  (do
+                            (fseek finput 0 SEEK-END)
+                            (let-do [ flength (ftell finput) ]
+                              (rewind finput)   ; aka (fseek ofile 0 SEEK-SET) 
+                              flength ))
+                buffer  (String.allocate length \0 )  ]  
+          (if (String.not-allocated? &buffer)
```

Comment:
why not use `null?` here?

Reply:
OK,fair enough. But better call it just `allocated?` then and wrap the form in `not`.

Reply:
Done.